### PR TITLE
Only manage internal DNS zone if configuration has been specified

### PR DESCRIPTION
**Pull request by mellowplace**
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/kops/5375)

### CODE REVIEW

**Comment by mellowplace** [comment]
MR with proposed fix here https://github.com/kubernetes/kops/pull/5375

File: pkg/apis/kops/v1alpha1/zz_generated.conversion.go [hunk 1]
```diff
@@ -2662,14 +2662,10 @@ func autoConvert_kops_LoadBalancerAccessSpec_To_v1alpha1_LoadBalancerAccessSpec(
 	out.Type = LoadBalancerType(in.Type)
 	out.IdleTimeoutSeconds = in.IdleTimeoutSeconds
 	out.AdditionalSecurityGroups = in.AdditionalSecurityGroups
+	// WARNING: in.UseApiInternal requires manual conversion: does not exist in peer-type
```
Comment (justinsb) [inline pkg/apis/kops/v1alpha1/zz_generated.conversion.go]: Ah - this means you need to add the new field to v1alpha1 also...

File: pkg/apis/kops/cluster.go [hunk 2]
```diff
@@ -286,6 +286,7 @@ type LoadBalancerAccessSpec struct {
 	Type                     LoadBalancerType `json:"type,omitempty"`
 	IdleTimeoutSeconds       *int64           `json:"idleTimeoutSeconds,omitempty"`
 	AdditionalSecurityGroups []string         `json:"additionalSecurityGroups,omitempty"`
+	UseApiInternal           bool             `json:"useApiInternal,omitempty"`
```
Comment (justinsb) [inline pkg/apis/kops/cluster.go]: Naming nit: it isn't very obvious (to me) what this does from the name.  How about `useForInternalApi` or `inClusterApi` or `useInsideCluster` ?

(And we have to take another pass to add this to v1alpha1 anyway ... :-) )

**Comment by justinsb** [comment]
One nit on the naming, but then I think we can get this in.  Sorry it took so long for me to get to this one...

If we're quick we can get this into 1.10...

File: pkg/apis/kops/v1alpha1/zz_generated.conversion.go (hunk 1 above)
Comment (mellowplace) [reply pkg/apis/kops/v1alpha1/zz_generated.conversion.go]: ah ok, I wasn't sure if v1alpha1 was an API that should not change and what I was supposed to do was add the field only to v1alpha2 and have a conversion function.  Will add it here as well.

File: pkg/apis/kops/cluster.go (hunk 2 above)
Comment (mellowplace) [reply pkg/apis/kops/cluster.go]: yep fair enough, I wasn't too keen on the name myself, I think `useForInternalApi` is clearer

**Comment by justinsb** [comment]
Thanks for the tweak @mellowplace 

/approve
/lgtm

**Pull request merged**